src/Illuminate/Routing/ImplicitRouteBinding.php

```diff
@@ -66,6 +88,26 @@ public static function resolveForRoute($container, $route)
         }
     }
 
+    /**
+     * Handle a missing parameter for a route binding.
+     *
+     * @param  \ReflectionParameter  $parameter
+     */
+    protected static function handleMissingParameter($parameter)
+    {
+        if (
+            ! $parameter->isDefaultValueAvailable() &&
+            static::$shouldPreventModelParameterMismatch &&
```

Comment:
Maybe better to start the if with `static::$shouldPreventModelParameterMismatch` to avoid performance hits if this is functionality is disabled